be/src/storage/tablet.h

```diff
@@ -173,6 +174,18 @@ class Tablet : public BaseTablet {
     void release_cumulative_lock() { _cumulative_lock.unlock(); }
     std::mutex& get_cumulative_lock() { return _cumulative_lock; }
 
+    // update compaction lock
+    void obtain_update_compaction_lock() {
```

Comment:
why not use the same lock as other type of tablets?

Reply:
Update compaction can only happens in update table, so here i put the `_update_compaction_lock` in the `tablet_updates` class (not in the `tablet` class).  
If the tablet is not from a update table, it will not have `_update_compaction_lock` field and maybe can save a little bit memory. Also, there is no meaning to acquire a  `_update_compaction_lock` lock from normal(not update) table.

Reply:
In `TabletUpdates::compaction` function, it used `_compaction_running.compare_exchange_strong` to check if there is another compaction is running. 
I think it is enough to avoid concurrent compaction. So is there any need to add one more lock?